custom_components/pun_sensor/sensor.py

```diff
@@ -58,50 +65,48 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry,
     async_add_entities(entities, update_before_add=False)
 
 
-def decode_fascia(fascia: int) -> str | None:
-    if fascia == 3:
-        return "F3"
-    elif fascia == 2:
-        return "F2"
-    elif fascia == 1:
-        return "F1"
-    else:
-        return None
-
-
-def fmt_float(num: float):
-    """Formatta adeguatamente il numero decimale"""
+def fmt_float(num: float) -> str | float:
+    """Formatta adeguatamente il numero decimale."""
     if has_suggested_display_precision:
         return num
-    
+
     # In versioni precedenti di Home Assistant che non supportano
     # l'attributo 'suggested_display_precision' restituisce il numero
     # decimale già adeguatamente formattato come stringa
-    return format(round(num, 6), '.6f')
+    return format(round(num, 6), ".6f")
+
 
 class PUNSensorEntity(CoordinatorEntity, SensorEntity, RestoreEntity):
-    """Sensore PUN relativo al prezzo medio mensile per fasce"""
+    """Sensore PUN relativo al prezzo medio mensile per fasce."""
 
-    def __init__(self, coordinator: PUNDataUpdateCoordinator, tipo: int) -> None:
+    def __init__(self, coordinator: PUNDataUpdateCoordinator, fascia: Fascia) -> None:
         super().__init__(coordinator)
 
         # Inizializza coordinator e tipo
         self.coordinator = coordinator
-        self.tipo = tipo
+        self.fascia = fascia
+
+        # BREAKING CHANGE, NEW ENTITY_ID CONVENTION FOR SENSORS
+        # non so come reagisce HA ad un cambio degli id
```

Comment:
HA orfana i sensori se cambia l'id, per quanto comodo in termini di generazione dei nomi e' un breaking change che fa' perdere i dati storici quindi non e' fattibile, commento da rimuovere.

si puo' fare un loop per generare gli stessi nomi ma non e' importante per ora.